lib/proxy/router.go

```diff
@@ -276,6 +276,7 @@ func (r *Router) DialHost(ctx context.Context, clientSrcAddr, clientDstAddr net.
 				if err != nil {
 					return nil, trace.Wrap(err)
 				}
+				principals = append(principals, fmt.Sprintf("%s.%s", host, clusterName))
```

Comment:
Should we just always have both `host` and `host + "." + clusterName` in the initial principal list rather than adding the principal here for openssh nodes? What's the difference between openssh nodes and non-openssh nodes accessed through proxy recording mode?

Reply:
I think you are correct, [this comment](https://github.com/gravitational/teleport/issues/36801#issuecomment-1895755864) seems to suggest proxy recording mode would also be affected